alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/LicenseMetrics.java

```diff
@@ -0,0 +1,118 @@
+package eu.xenit.alfred.telemetry.binder;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import org.alfresco.service.descriptor.Descriptor;
+import org.alfresco.service.descriptor.DescriptorService;
+import org.alfresco.service.license.LicenseDescriptor;
+import org.alfresco.service.license.LicenseService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import javax.annotation.Nonnull;
+
+public class LicenseMetrics implements MeterBinder, ApplicationContextAware {
+
+    private static final Logger logger = LoggerFactory.getLogger(LicenseMetrics.class);
+
+    private static final String METRIC_NAME_LICENSE = "license";
+
+    private ApplicationContext ctx;
+    private DescriptorService descriptorService;
+
+    public LicenseMetrics(DescriptorService descriptorService) {
+        this.descriptorService = descriptorService;
+    }
+
+    @Override
+    public void bindTo(@Nonnull MeterRegistry registry) {
+        // do not do anything for Community
+        Descriptor serverDescriptor = descriptorService.getServerDescriptor();
+        if(!"Enterprise".equals(serverDescriptor.getEdition())) {
+            logger.info("Edition={}, license metrics are not available", serverDescriptor.getEdition());
+            return;
+        }
+
+        Gauge.builder(METRIC_NAME_LICENSE + ".valid", ctx, LicenseMetrics::getValid)
+                .description("Whether the license is still valid")
+                .register(registry);
+
+        Gauge.builder(METRIC_NAME_LICENSE + ".remainingDays", descriptorService, LicenseMetrics::getRemainingDays)
+                .description("Remaining days")
+                .register(registry);
+        Gauge.builder(METRIC_NAME_LICENSE + ".maxDocs", descriptorService, LicenseMetrics::getMaxDocs)
+                .description("Max docs")
+                .register(registry);
+        Gauge.builder(METRIC_NAME_LICENSE + ".maxUsers", descriptorService, LicenseMetrics::getMaxUsers)
+                .description("Max users")
+                .register(registry);
+        Gauge.builder(METRIC_NAME_LICENSE + ".isClusterEnabled", descriptorService, LicenseMetrics::isClusterEnabled)
+                .description("Clustering enabled")
+                .register(registry);
+        Gauge.builder(METRIC_NAME_LICENSE + ".isCryptodocEnabled", descriptorService, LicenseMetrics::isCryptodocEnabled)
+                .description("Encription enabled")
+                .register(registry);
+        Gauge.builder(METRIC_NAME_LICENSE + ".isHeartbeatDisabled", descriptorService, LicenseMetrics::isHeartbeatDisabled)
```

Comment:
Maybe we should use the `-Enabled` variant for all those metrics. Switching between 'enabled' and 'disabled' is a bit confusing.

Reply:
I did not add reviewers yet, added now.

I can follow Alfresco's reasoning: anything different from default situation (non-clustered, non-encrypted, with heartbeat) gets a boolean function with value 'true'. 
For me it is not confusing, but if other people think it is, I will change the names. What do you think @kerkhofsd and @vierbergenlars?

Reply:
I'm fine with the current naming, but wouldn't be opposed to making them all `-Enabled` either.

Reply:
For consistency, my preference goes to an `-Enabled` variant. 
Furthermore, my preference would also be to make the metrics more structured. Something like: `license.heartbeat.enabled`. Should we need / want to add another heartbeat related metric later, we could easily group it under the same comprehensive prefix.

Reply:
This is better, I will implement it.